code/game/machinery/scan_gate.dm

```diff
@@ -0,0 +1,226 @@
+#define SCANGATE_NONE 			"Off"
+#define SCANGATE_MINDSHIELD 	"Mindshield"
+#define SCANGATE_NANITES 		"Nanites"
+#define SCANGATE_DISEASE 		"Disease"
+#define SCANGATE_GUNS 			"Guns"
+#define SCANGATE_WANTED			"Wanted"
+#define SCANGATE_SPECIES		"Species"
+
+
+/obj/machinery/scanner_gate
+	name = "scanner gate"
+	desc = "A gate able to perform mid-depth scans on any organisms who pass under it."
+	icon = 'icons/obj/machines/scangate.dmi'
+	icon_state = "scangate"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 50
+	circuit = /obj/item/circuitboard/machine/scanner_gate
+	var/scanline_timer
+	var/next_beep = 0 //avoids spam
+
+	var/locked = FALSE
+	var/scangate_mode = SCANGATE_NONE
+	var/disease_threshold = DISEASE_SEVERITY_MINOR
+	var/nanite_cloud = 0
+	var/datum/species/detect_species = /datum/species/human
+	var/reverse = FALSE //If true, signals if the scan returns false
+
+/obj/machinery/scanner_gate/Initialize()
+	. = ..()
+	set_scanline("passive")
+
+/obj/machinery/scanner_gate/examine(mob/user)
+	..()
+	if(locked)
+		to_chat(user, "<span class='notice'>The control panel is ID-locked. Swipe a valid ID to unlock it.</span>")
+	else
+		to_chat(user, "<span class='notice'>The control panel is unlocked. Swipe an ID to lock it.</span>")
+
+/obj/machinery/scanner_gate/Crossed(atom/movable/AM)
+	..()
+	if(!(stat & (BROKEN|NOPOWER)) && isliving(AM))
+		perform_scan(AM)
+
+/obj/machinery/scanner_gate/proc/set_scanline(type, duration)
+	cut_overlays()
+	deltimer(scanline_timer)
+	add_overlay(type)
+	if(duration)
+		scanline_timer = addtimer(CALLBACK(src, .proc/set_scanline, "passive"), duration, TIMER_STOPPABLE)
```

Comment:
Is there something special simple flick or state change won't do here ? Cutting all overlays seems excessive.

Reply:
I guess i could store the scanline overlay in a var, i was following the example of other update_icon procs

Reply:
It's not horrible, just could be optimized probably.